src/System.Net.Security/src/System/Net/NegotiateStreamPal.Windows.cs

```diff
@@ -0,0 +1,458 @@
+            Interop.SecurityStatus winStatus = (Interop.SecurityStatus)SSPIWrapper.InitializeSecurityContext(
+                GlobalSSPI.SSPIAuth,
+                credentialsHandle,
+                ref securityContext,
+                spn,
+                ContextFlagsAdapterPal.GetInteropFromContextFlagsPal(requestedContextFlags),
+                Interop.SspiCli.Endianness.Network,
+                inSecurityBufferArray,
+                outSecurityBuffer,
+                ref outContextFlags);
+
+            contextFlags = ContextFlagsAdapterPal.GetContextFlagsPalFromInterop(outContextFlags);
+            return SecurityStatusAdapterPal.GetSecurityStatusPalFromInterop(winStatus);
+        }
+
+        internal static SecurityStatusPal CompleteAuthToken(
+            ref SafeDeleteContext securityContext,
+            SecurityBuffer[] inSecurityBufferArray)
+        {
+            Interop.SecurityStatus winStatus = (Interop.SecurityStatus)SSPIWrapper.CompleteAuthToken(
+                GlobalSSPI.SSPIAuth,
+                ref securityContext,
+                inSecurityBufferArray);
+            return SecurityStatusAdapterPal.GetSecurityStatusPalFromInterop(winStatus);
+        }
+
+        internal static SecurityStatusPal AcceptSecurityContext(
+            SafeFreeCredentials credentialsHandle,
+            ref SafeDeleteContext securityContext,
+            ContextFlagsPal requestedContextFlags,
+            SecurityBuffer[] inSecurityBufferArray,
+            SecurityBuffer outSecurityBuffer,
+            ref ContextFlagsPal contextFlags)
+        {
+            Interop.SspiCli.ContextFlags outContextFlags = Interop.SspiCli.ContextFlags.Zero;
+            Interop.SecurityStatus winStatus = (Interop.SecurityStatus)SSPIWrapper.AcceptSecurityContext(
+                GlobalSSPI.SSPIAuth,
+                credentialsHandle,
+                ref securityContext,
+                ContextFlagsAdapterPal.GetInteropFromContextFlagsPal(requestedContextFlags),
+                Interop.SspiCli.Endianness.Network,
+                inSecurityBufferArray,
+                outSecurityBuffer,
+                ref outContextFlags);
+
+            contextFlags = ContextFlagsAdapterPal.GetContextFlagsPalFromInterop(outContextFlags);
+            return SecurityStatusAdapterPal.GetSecurityStatusPalFromInterop(winStatus);
+        }
+
+        internal static void ValidateImpersonationLevel(TokenImpersonationLevel impersonationLevel)
+        {
+            if (impersonationLevel != TokenImpersonationLevel.Identification &&
+                impersonationLevel != TokenImpersonationLevel.Impersonation &&
+                impersonationLevel != TokenImpersonationLevel.Delegation)
+            {
+                throw new ArgumentOutOfRangeException("impersonationLevel", impersonationLevel.ToString(), SR.net_auth_supported_impl_levels);
+            }
+        }
+
+        internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
```

Comment:
If both PAL implementations are emitting Win32Exception, and the caller expected Win32Exception, why did this get dropped to Exception?

(It'd make more sense if it turned some of them into OutOfMemoryException, or other types)